src/ngraph/runtime/gpu/cudnn_emitter.cpp

```diff
@@ -236,44 +236,80 @@ size_t runtime::gpu::CUDNNEmitter::build_reduce_forward(const cudnnReduceTensorO
+                            runtime::gpu::invoke_memory_primitive(m_ctx, workspace_idx);
+                        void* reduce_buffer =
+                            runtime::gpu::invoke_memory_primitive(m_ctx, reduce_buffer_idx);
+                        CUDNN_SAFE_CALL(cudnnReduceTensor(*m_ctx->cudnn_handle,
+                                                          desc,
+                                                          workspace_indices_ptr,
+                                                          indices_size,
+                                                          workspace_ptr,
+                                                          workspace_size,
+                                                          alpha,
+                                                          input_desc,
+                                                          inputs[0],
+                                                          beta,
+                                                          output_desc,
+                                                          reduce_buffer));
+                        gpu::invoke_primitive(m_ctx, convert_idx, &workspace_indices_ptr, outputs);
+                        debug_sync();
+                    }});
+            }
+            else
+            {
+                reduce.reset(new gpu::primitive{
+                    [=, &desc, &input_desc, &output_desc](void** inputs, void** outputs) {
+
+                        void* workspace_ptr =
+                            runtime::gpu::invoke_memory_primitive(m_ctx, workspace_idx);
+                        void* reduce_buffer =
+                            runtime::gpu::invoke_memory_primitive(m_ctx, reduce_buffer_idx);
+                        CUDNN_SAFE_CALL(cudnnReduceTensor(*m_ctx->cudnn_handle,
+                                                          desc,
+                                                          outputs[0],
+                                                          indices_size,
+                                                          workspace_ptr,
+                                                          workspace_size,
+                                                          alpha,
+                                                          input_desc,
+                                                          inputs[0],
+                                                          beta,
+                                                          output_desc,
+                                                          reduce_buffer));
+                        debug_sync();
+                    }});
+            }
+        }
+        else
         {
             std::stringstream ss_er;
-            ss_er
-                << "Unsupported Type: Only uint32 currently supported for indices in op ArgReduce ";
+            ss_er << "Unsupported Type: Only uint32 & uint64 currently supported for indices in op "
```

Comment:
nit: add a `output_type.c_type_string()` in error message, so we know what's is required.